### ISSUE TITLE
Disable MVC routes for Angular application

### ISSUE THREAD

**Issue by olicooper** [abpframework/abp#2604]
When we use the abp cli to create an app with the angular ui, we can still access the MVC routes such as `localhost:44344/Account/Register` and `localhost:44344/Account/Manage`.

Is this by design? If not, could we remove it from the cli generated files?

I've seen we can manually disable the routes by removing `AbpAspNetCoreMvcUiBasicThemeModule` from the `HttpApi.Host` module:
```csharp
[DependsOn(
    ...
    typeof(AbpAspNetCoreMvcUiBasicThemeModule),
    ...
    )]
```
But `wwwroot` still has redundant files in it by default.

**Comment by olicooper** [comment]
It seems some of the localization disappears when removing the dependency form the module so it isn't as easy as I initially thought. I have found it can be solved two ways.
Depend on `typeof(AbpAspNetCoreMvcUiMultiTenancyModule)` instead (but this includes style bundling which isn't required by angular)
Or add the localization manually by adding these into the relevant sections of the `HttpApi.Host` module (taken from the [AbpAspNetCoreMvcUiMultiTenancyModule](https://github.com/abpframework/abp/blob/dev/framework/src/Volo.Abp.AspNetCore.Mvc.UI.MultiTenancy/Volo/Abp/AspNetCore/Mvc/UI/MultiTenancy/AbpAspNetCoreMvcUiMultiTenancyModule.cs) module):
```csharp
Configure<AbpVirtualFileSystemOptions>(options =>
{
    options.FileSets.AddEmbedded<AbpAspNetCoreMvcUiMultiTenancyModule>();
});
```
```csharp
Configure<AbpLocalizationOptions>(options =>
{
    options.Resources
        .Add<AbpUiMultiTenancyResource>("en")
        .AddVirtualJson("/Volo/Abp/AspNetCore/Mvc/UI/MultiTenancy/Localization");
});
```

**Comment by maliming** [comment]
> Is this by design? If not, could we remove it from the cli generated files?

Because the project contains an identity server, it may needs these pages.

You can use the `--separate-identity-server` option when creating angular projects.

```
angular: Angular. There are some additional options for this template:
--separate-identity-server: Separates the identity server application from the API host application. If not specified, you will have a single endpoint in the server side.
```

**Comment by olicooper** [comment]
Thanks for your comment. I will try to look at this when I can.
Unfortunately the only immediate solution for me (I haven't tested yours yet due to time constraints), is to depend on `typeof(AbpAspNetCoreMvcUiMultiTenancyModule)` because my other solution made a number of multi-tenant endpoints not work (see image below) due to the multi-tenant controller not being loaded.
![image](https://user-images.githubusercontent.com/22296715/72167261-bc0f1b00-33c2-11ea-943b-16c3cb15cdc2.png)

changing the module dependency seems to remove the MVC routes but keep the rest so I will see how it goes.

**Comment by olicooper** [comment]
Now that https://github.com/abpframework/abp/issues/2512 is released in abp v2.0 and registration can be disabled, I think this ticket can be closed.
